crates/deltalake-core/src/schema/partitions.rs

```diff
@@ -40,6 +42,20 @@ pub struct PartitionFilter {
     pub value: PartitionValue,
 }
 
+fn parse_timestamp(timestamp_str: &str) -> Option<NaiveDateTime> {
+    // Timestamp formats as per https://github.com/delta-io/delta/blob/master/PROTOCOL.md#partition-value-serialization
+    let formats = &["%Y-%m-%d %H:%M:%S", "%Y-%m-%d %H:%M:%S%.6f"];
```

Comment:
I believe you can leave just this format, as it will successfully capture both cases (with or without microseconds).
Moreover, you can omit the "6" constraint (i.e. "%Y-%m-%d %H:%M:%S%.f") because it does not add anything - timestamp strings that include more/less/no microseconds will still be successfully parsed, both with "%.6f" and with "%.f".

Reply:
Done